generics/helpers/programSolutionUtilities.js

```diff
@@ -0,0 +1,239 @@
+ * @param {Object} bodyData - Request body containing scope data (e.g., entities, organizations).
+ * @param {String} tenantId - Tenant ID for validation and context.
+ * @param {String} orgId - Organization ID for validation context.
+ * @param {Object} userDetails - Object containing user roles and authentication details.
+ * @returns {Promise<Object>} - Returns an object with updateObject and success flag, or an error with status and message.
+ */
+function getUpdateObjectTOAddScope(bodyData, tenantId, orgId, userDetails) {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const ALL_SCOPE_VALUE = messageConstants.common.ALL_SCOPE_VALUE;
+      // Setup for MongoDB update operation using $addToSet
+      let updateObject = { $addToSet: {} };
+      let validationExcludedEntitiesKeys = [];
+      let tenantDetails;
+      let adminTenantAdminRole = [messageConstants.common.ADMIN, messageConstants.common.TENANT_ADMIN];
+
+      // Check if user is Admin or Tenant Admin
+      if (gen.utils.validateRoles(userDetails.roles, adminTenantAdminRole)) {
+        // Fetch tenant details to validate org codes and scope keys
+        tenantDetails = await userService.fetchTenantDetails(tenantId, userDetails.userToken);
+        if (!tenantDetails?.success || !tenantDetails?.data?.meta) {
+          throw {
+            message: messageConstants.apiResponses.FAILED_TO_FETCH_TENANT_DETAILS,
+            status: httpStatusCode.bad_request.status,
+          };
+        }
+
+        // Store validation-excluded scope keys if present
+        if (
+          Array.isArray(tenantDetails?.data?.meta?.validationExcludedScopeKeys) &&
+          tenantDetails.data.meta.validationExcludedScopeKeys.length > 0
+        ) {
+          // Fetch tenant details (will include valid org codes & validationExcludedScopeKeys)
+          validationExcludedEntitiesKeys.push(...tenantDetails.data.meta.validationExcludedScopeKeys);
+        }
+
+        // Handle organization values if passed
+        if (bodyData.organizations) {
+          if (Array.isArray(bodyData.organizations)) {
+            if (bodyData.organizations.includes(ALL_SCOPE_VALUE)) {
+              // Add "ALL" if specified
+              updateObject.$addToSet[`scope.organizations`] = { $each: [ALL_SCOPE_VALUE] };
+            } else {
+              const validOrgCodes = tenantDetails.data.organizations.map((org) => org.code);
+              const isValid = bodyData.organizations.every((orgCode) => validOrgCodes.includes(orgCode));
+              if (!isValid) {
+                throw {
+                  message: messageConstants.apiResponses.INVALID_ORGANIZATION,
+                  status: httpStatusCode.bad_request.status,
+                };
+              }
+              updateObject.$addToSet[`scope.organizations`] = { $each: bodyData.organizations };
+            }
+          }
+        }
+      }
+
+      // This logic we need to re-look --------------------------------------------
```

Comment:
@MallanagoudaB remove this from this function. if you want you can keep the commented code in helper function for future reference. if this is there in project service PR please correct there as well. please add a rply to this comment